buffer/src/test/java/io/netty/buffer/PoolSubpageReleaseTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.buffer;
+
+import org.junit.Test;
+import org.junit.platform.commons.util.ReflectionUtils;
+
+import java.lang.reflect.Field;
+
+import static org.junit.Assert.assertEquals;
+
+public class PoolSubpageReleaseTest {
+    public PoolSubpageReleaseTest() {
+    }
+
+    @Test
+    public void testAbnormalPoolSubpageRelease() {
+        int elemSize = 8192;
+        run(elemSize);
+    }
+
+    @Test
+    public void testNormalPoolSubpageRelease() {
+        // 0 < eleSize <= 28672, != 8192
+        int elemSize = 1234;
+        run(elemSize);
+    }
+
+    private void run(int elemSize) {
+        int oneLength = 128, fixLength = 256;
+        ByteBuf[] ones = new ByteBuf[oneLength];
+        ByteBuf[] fixByteBuf = new ByteBuf[fixLength];
+
+        //allocate
+        for (int i = 0; i < fixLength; i++) {
+            fixByteBuf[i] = ByteBufAllocator.DEFAULT.buffer(elemSize, elemSize);
```

Comment:
You should directly instance a PooledByteBufAllocator here... or even merge this into PooledByteBufAllocatorTest